router/network.go

```diff
@@ -124,6 +126,22 @@ func (network *netHandle) SendPost(ctx context.Context, structData integrations.
 					formValues.Set(key, fmt.Sprint(val)) // transformer ensures top level string values, still val.(string) would be restrictive
 				}
 				payload = strings.NewReader(formValues.Encode())
+			case "GZIP":
+				strValue, ok := bodyValue["payload"].(string)
+				if !ok {
+					return &utils.SendPostResponse{
+						StatusCode:   400,
+						ResponseBody: []byte("400 Unable to construct gzip payload. Unexpected transformer response"),
+					}
+				}
+				zippedData, err := base64.StdEncoding.DecodeString(strValue)
```

Comment:
why do we need to base64 ?

Reply:
Transmitting GZIP-compressed binary data from the transformer to the server can introduce compatibility issues. By encoding the payload at the transformer's end and decoding it at the server's end, ensures seamless data transfer without compatibility concerns.
cc- @koladilip

Reply:
Do we really need to transmit binary data from transformer to rudder-server? What is the use case? If we can skip gzip between transformer-server communication it would be cleaner.

Reply:
Use case is to send the gzipped data to mixpanel for a specific batch endpoint [/import ](https://developer.mixpanel.com/reference/import-events)

Reply:
@Gauravudia we can do that too.

Reply:
@atzoum Removed the base64 and added the gzip compression in server